src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -288,6 +288,9 @@ public void SslStream_StreamToStream_Write_ReadByte_Success()
                 bool result = DoHandshake(clientSslStream, serverSslStream);
                 Assert.True(result, "Handshake completed.");
 
+                // Ensure Server's ReadAsync task doesn't timeout before WriteAsync completes.
+                network.WaitForReadDataTimeoutMilliseconds = 2 * TestConfiguration.PassingTestTimeoutMilliseconds;
```

Comment:
Can you please explain the logic here?

Reply:
This value ```WaitForReadDataTimeoutMilliseconds``` is the amount of time the ReadAsync task waits to receive data from the server. I want this value to be greater than the subsequent WriteAsync task's completion time. So that if the WriteAsync later fails or hangs it will timeout after TestConfiguration.PassingTestTimeoutMilliseconds elapses. By setting the ReadAsync value 2 times greater, I ensure it is the WriteAsync tasks that timesout and not the ReadAsync.

Reply:
One thing, the lowest timeout was 20seconds in the old way. That still seems like a lot of time to have to wait, seeing as it is a virtual network adapter... The PR #25297 now has me wondering.

Reply:
That made no difference, (the init crypto) if the test change makes sense, then I guess that's it, The timeout seems like it shouldn't be hit but I am not sure of the perf/timing of the test servers so I will leave up to the MS people :)

Reply:
> This value WaitForReadDataTimeoutMilliseconds is the amount of time the ReadAsync task waits to receive data from the server. I want this value to be greater than the subsequent WriteAsync task's completion time. So that if the WriteAsync later fails or hangs it will timeout after TestConfiguration.PassingTestTimeoutMilliseconds elapses. By setting the ReadAsync value 2 times greater, I ensure it is the WriteAsync tasks that timesout and not the ReadAsync.

I don't understand.  Which WriteAsync is failing?  We should never have tests that rely on waiting for 30-60 second timeouts.

Reply:
I explained what the test is doing in the comment [here](https://github.com/dotnet/corefx/pull/25261#issuecomment-344837578), the WriteAsync I am referring to is the [here](https://github.com/Priya91/corefx-1/blob/97a227fd722b823d8aff2b3a0ef1120b7680f9a2/src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs#L309)

Reply:
> Which WriteAsync is failing?

The WriteAsync shouldn't fail, that's what the test is supposed to verify. The timeout that was set on this WriteAsync task to verify it doesn't fail was 4 * 60 secs, and the timeout set on the previous readasync was 30s. In some cases maybe when the the number of tests increased and the number of threads per process increased, the WriteAsync task could be delayed, and it took more than 30s to finish, by that time the already started ReadAsync timed out and caused the test failure.

Reply:
> In some cases maybe when the the number of tests increased and the number of threads per process increased, the WriteAsync task could be delayed, and it took more than 30s to finish, by that time the already started ReadAsync timed out and caused the test failure.

Even in stress situations, it shouldn't take anywhere near 30s to complete; it should be close to instantaneous.  Something else must be going on.

Reply:
> Something else must be going on.

yeah, i've mentioned this in the initial comment here,

> Or if there isn't a hang, then something has changed that is causing the async operations take more time to finish than previously, which again is in the Encrypt/Decrypt part (ReadAsync/WriteAsync) code path, thus requires more investigation to understand the behavior.

I'm not sure how to proceed to debug this part, I've captured all this data in the issue #19705

Reply:
I think I have solved this problem, you won't need to make this change in the tests.